src/main/scala/chisel3/util/Counter.scala

```diff
@@ -59,4 +70,18 @@ object Counter
     when (cond) { wrap = c.inc() }
     (c.value, cond && wrap)
   }
+
+  /** Instantiate a [[Counter! counter]] with the specified number of counts, a gate, and a restart.
+   */
+  def apply(cond: Bool, n: Int, restart: Bool): (UInt, Bool) = apply(cond, n, restart, 0)
+
+  /** Instantiate a [[Counter! counter]] with the specified number of counts, a gate, a restart, and an init.
+   */
+  def apply(cond: Bool, n: Int, restart: Bool, init: Int): (UInt, Bool) = {
+    val c = new Counter(n, init)
+    var wrap: Bool = null
```

Comment:
Avoid nulls and vars. This is just bad.

Perhaps add a wrap signal to the Counter object which can be accessed outside `.inc()`?